feat: Release notes for Codacy Cloud September 2022 (#1441)
codacy/docs: docs/release-notes/cloud/cloud-2022-09.md

included_jira_versions: ['2022.Q3.7']
codacy_tools_version_old: https://github.com/codacy/codacy-tools/releases/tag/6.2.59
codacy_tools_version_new: https://github.com/codacy/codacy-tools/releases/tag/6.3.14
---

# Cloud September 2022

These release notes are for the Codacy Cloud updates during September 2022.

📢 [Visit the Codacy roadmap](https://roadmap.codacy.com) and <span class="skip-vale">let us know</span> your feedback on both new and planned product updates!

## Product enhancements

-   Codacy now displays the coverage variation metric with a precision of two decimal places on the [Repository Dashboard](../../repositories/repository-dashboard.md) and [Organization Overview](../../organizations/organization-overview.md), and you can [define quality gates](../../repositories-configure/adjusting-quality-settings.md#gates) with a coverage variation threshold using the same precision.

    The increased precision of the metric reflects the code coverage changes better by reducing the false changes due to rounding errors. (IO-54, IO-56)

    ![Coverage threshold with two decimal places on the Quality settings page](../images/io-56.png)

> -   While [configuring a coding standard](../../organizations/using-a-coding-standard.md), there's now a checkbox on the header of the code pattern list that allows toggling all code patterns that are currently visible on the list. This allows you to conveniently toggle code patterns in bulk, for example, to enable all security code patterns. (CY-6336)

Reviewer: Suggestion to focus on the new capability instead of the new checkbox:

```suggestion
-   While [configuring a coding standard](../../organizations/using-a-coding-standard.md), you can now toggle all code patterns that are currently visible on the list using the new checkbox on the header. This allows you to conveniently toggle code patterns in bulk, for example, to enable all security code patterns. (CY-6336)
```

Author: Good idea, thanks @claudiacarpinteiro!

    ![Selecting all code patterns while configuring a coding standard](../images/cy-6336.png)

-   Now, the **Files** tab on [pull request](../../repositories/pull-requests.md#files-tab) and [commit](../../repositories/commits.md#files-tab) pages displays only files with a variation of their code quality metrics, to make it more convenient to analyze the impact of your changes on the code quality of the files.

    You can still list all files that the pull request updated, even if their code quality metrics didn't change, by toggling the new option **Show files without code quality changes**. (CY-6455)

    ![Updated Files tab focusing only on files that have a variation in the code quality metrics](../images/cy-6455.png)

-   The **Repositories list** page has improved loading times using a short-lived cache in the user's browser. (CY-6421)

## Bug fixes

> -   Fixed an issue that could cause the commit pages to display an empty **Diff** tab. (IO-129)

Reviewer 2: I wouldn't include this just because the issue was also introduced by a task done in the same month. It was caused by this: https://codacy.atlassian.net/browse/CY-6411

Author: Thanks, @inesgomas! Indeed, I was unsure if this issue had existed for a long time or not, but I couldn't find out on IO-129. I'll remove this release note.

Reviewer 2: Yeah we didn't link the tasks 🙈
